CORENET-6203: Enable test for LGW L2

arkadeepsen opened the pull request:
This PR enables the test for LGW L2 which was previously disabled.

arkadeepsen commented:
> @arkadeepsen also we need this merged for the tests here to be able to pass [openshift/ovn-kubernetes#2676](https://github.com/openshift/ovn-kubernetes/pull/2676)

@jcaamano Ack. However, I think there is some issue with the sniffer daemonsets. The test is trying to deploy it twice without removing it. Either we should remove it and then deploy it again, OR, we should deploy it only once.

jcaamano commented:
> > @arkadeepsen also we need this merged for the tests here to be able to pass [openshift/ovn-kubernetes#2676](https://github.com/openshift/ovn-kubernetes/pull/2676)
> 
> @jcaamano Ack. However, I think there is some issue with the sniffer daemonsets. The test is trying to deploy it twice without removing it. Either we should remove it and then deploy it again, OR, we should deploy it only once.

The VRF-Lite test case you mean?

jcaamano commented:
/lgtm cancel

due to https://github.com/openshift/origin/pull/30009#issuecomment-3107852989

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented:
> > > > @arkadeepsen also we need this merged for the tests here to be able to pass [openshift/ovn-kubernetes#2676](https://github.com/openshift/ovn-kubernetes/pull/2676)
> > > 
> > > 
> > > @jcaamano Ack. However, I think there is some issue with the sniffer daemonsets. The test is trying to deploy it twice without removing it. Either we should remove it and then deploy it again, OR, we should deploy it only once.
> > 
> > 
> > The VRF-Lite test case you mean?
> 
> Yeah, for the VRF-Lite test case I think you are right. I thought I did a better job there 🤦
> 
> So as we call `beforeEach` we should also call `afterEach`. Additionally, we would need to delete `snifferNamespace`. Potentially, also delete either things within `targetNamespace`, or delete that namespace itself and replace it with another one tricking the e2e framework.

Added a call to `afterEach` at the end of `test` function. Removed the packet sniffer daemonset in the `afterEach` function. This has fixed the issue being faced earlier. Will wait for https://github.com/openshift/ovn-kubernetes/pull/2676 to merge to get a green signal on the bgp test.

arkadeepsen commented:
/retest-required

arkadeepsen commented:
/retest-required

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

arkadeepsen commented on test/extended/networking/route_advertisements.go:
```diff
@@ -472,6 +465,7 @@ var _ = g.Describe("[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:Ro
 				errors.Join(err, runOcWithRetryIgnoreOutput(oc.AsAdmin().WithoutNamespace(), "delete", "ra", testCUDNName))
 				errors.Join(err, runOcWithRetryIgnoreOutput(oc.AsAdmin().WithoutNamespace(), "delete", "clusteruserdefinednetwork", testCUDNName))
 				errors.Join(err, deallocateSubnets(oc, userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet))
+				errors.Join(err, runOcWithRetryIgnoreOutput(oc.AsAdmin(), "delete", "ds", packetSnifferDaemonSet.Name, "-n", snifferNamespace))
```
Pushed the suggested changes.

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

jcaamano commented:
/test e2e-metal-ipi-ovn-dualstack-bgp

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-local-gateway

arkadeepsen commented:
/test e2e-metal-ipi-ovn-dualstack-bgp-local-gw

kyrtapz commented:
/lgtm

arkadeepsen commented:
/cherry-pick release-4.19
